modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -172,10 +172,11 @@ private HdfsFile getHdfsFile(String resolvedFileName) {
     }
 
     private HdfsFile createHdfsFile(String resolvedFileName) {
-        HdfsFile hdfsFile = new HdfsFile();
         Path filePath = getFilePath(resolvedFileName);
-        hdfsFile.setPath(filePath);
-        hdfsFile.setFsDataOutputStream(createFsDataOutputStream(hdfs, filePath));
+        FSDataOutputStream fsDataOutputStream = createFsDataOutputStream(hdfs, filePath);
+
+        HdfsFile hdfsFile = new HdfsFile(filePath, fsDataOutputStream);
```

Comment:
Actually for me the issue is not the `filePath` but the `FSDataOutputStream`. Because for a file to me it is natural to provide the path, not some magical stream (That should be the `HdfsFile` internal business.)
But for that I have to deal with append and archive (maybe just append). The actual purpose of this PR was not to refactor, and I made a cut here as for me it was acceptable (compared to the original.)